pkg/graveler/ref/manager.go

```diff
@@ -456,23 +487,38 @@ func (m *KVManager) GetCommitByPrefix(ctx context.Context, repository *graveler.
 }
 
 func (m *KVManager) GetCommit(ctx context.Context, repository *graveler.RepositoryRecord, commitID graveler.CommitID) (*graveler.Commit, error) {
+	key := fmt.Sprintf("%s:%s", repository.RepositoryID, commitID)
+	v, err := m.repoCache.GetOrSet(key, func() (v interface{}, err error) {
+		return m.getCommitBatch(ctx, repository, commitID)
```

Comment:
It won't cache NotFound commits. I can't remember if we expect this to happen a lot. The chances of a user pointing directly to a commit hash which wasn't created yet is not big. However, we accept refs which could be anything. If we currently check that `main` isn't a commit, I think we should avoid that constant call to the store.

Reply:
I think that full commit ID for non existing is something that probably will not be part of any main flow. Anything short or resolved from branch or tag should be an existing commit.
So filling the cache with non existing commit for a case that a user/app try to identify if commit exits I assume it will be something I don't want to return answer from the last 30sec.
Keeping it this way.

Reply:
@nopcoder missed this one?

Reply:
Will open an issue/pr to handle optimization of branch vs commit prefix.